### PR TITLE
Configure openshift-ci for shipwright-io/build

### PR DESCRIPTION
**Pull request by divyansh42**
Initial work done by @jitendar-singh in https://github.com/openshift/release/pull/39322

configuration files added to run unit-test, integration test and e2e test

- ci-operator/jobs/shipwright-io/build/shipwright-io-build-main-presubmits.yaml
- ci-operator/config/shipwright-io/build/shipwright-io-build-main.yaml
- core-services/prow/02_config/shipwright-io/build/_pluginconfig.yaml

### CODE REVIEW

File: ci-operator/config/shipwright-io/build/shipwright-io-build-main.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+binary_build_commands: make build
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.19
+images:
+- dockerfile_literal: |
+    # Dockerfile to bootstrap build and test in openshift-ci
+    FROM quay.io/openshift/origin-cli:4.14 AS base
```
Comment (xinredhat) [inline ci-operator/config/shipwright-io/build/shipwright-io-build-main.yaml:10]: this is not needed, when you want to run oc command at test step, just need to add `cli: 4.14`

File: core-services/image-mirroring/openshift/mapping_origin_4_14 [hunk 2]
```diff
@@ -294,6 +294,7 @@ registry.ci.openshift.org/origin/4.14:service-ca-operator quay.io/openshift/orig
 registry.ci.openshift.org/origin/4.14:service-catalog quay.io/openshift/origin-service-catalog:4.14 quay.io/openshift/origin-service-catalog:4.14.0 quay.io/openshift/origin-service-catalog:latest
 registry.ci.openshift.org/origin/4.14:special-resource-operator quay.io/openshift/origin-special-resource-operator:4.14 quay.io/openshift/origin-special-resource-operator:4.14.0 quay.io/openshift/origin-special-resource-operator:latest
 registry.ci.openshift.org/origin/4.14:special-resource-rhel8-operator quay.io/openshift/origin-special-resource-rhel8-operator:4.14 quay.io/openshift/origin-special-resource-rhel8-operator:4.14.0 quay.io/openshift/origin-special-resource-rhel8-operator:latest
+registry.ci.openshift.org/origin/4.14:src-kubectl quay.io/openshift/origin-src-kubectl:4.14 quay.io/openshift/origin-src-kubectl:4.14.0 quay.io/openshift/origin-src-kubectl:latest
```
Comment (coreydaley) [inline core-services/image-mirroring/openshift/mapping_origin_4_14:297]: I am 99% sure that this is not the image that we want to mirror, we only need it for running the tests.

Comment (divyansh42) [reply core-services/image-mirroring/openshift/mapping_origin_4_14:297]: Probably my understanding was wrong here, I thought this step was required for using the image in tests.
It was generated by the command `make openshift-image-mirror-mappings`

File: core-services/prow/02_config/shipwright-io/build/_pluginconfig.yaml [hunk 3]
```diff
@@ -14,6 +13,10 @@ external_plugins:
     events:
     - pull_request
     name: needs-rebase
+lgtm:
+- repos:
+  - shipwright-io/build
+  review_acts_as_lgtm: true
```
[the remark this answers is not in the document]
Comment (divyansh42) [reply core-services/prow/02_config/shipwright-io/build/_pluginconfig.yaml:19]: I am not sure about it, It just copied from the Jitendar's previous PR https://github.com/openshift/release/pull/39322

**Comment by coreydaley** [comment]
The two files that would not install are namespaced BuildStrategy, instead of ClusterBuildStrategy. I wonder if we need to exclude those for the purposes of testing?
```
Error from server (NotFound): error when creating "samples/buildstrategy/buildpacks-v3/buildstrategy_buildpacks-v3-heroku_namespaced_cr.yaml": namespaces "ci-op-ywry1nk6" not found
Error from server (NotFound): error when creating "samples/buildstrategy/buildpacks-v3/buildstrategy_buildpacks-v3_namespaced_cr.yaml": namespaces "ci-op-ywry1nk6" not found
```

**Comment by coreydaley** [comment]
We should probably just install a specific list of buildstrategies that are needed for the tests to run instead of installing everything from the samples folder.

**Comment by divyansh42** [comment]
> The two files that would not install are namespaced BuildStrategy, instead of ClusterBuildStrategy. I wonder if we need to exclude those for the purposes of testing?
> 
> ```
> Error from server (NotFound): error when creating "samples/buildstrategy/buildpacks-v3/buildstrategy_buildpacks-v3-heroku_namespaced_cr.yaml": namespaces "ci-op-ywry1nk6" not found
> Error from server (NotFound): error when creating "samples/buildstrategy/buildpacks-v3/buildstrategy_buildpacks-v3_namespaced_cr.yaml": namespaces "ci-op-ywry1nk6" not found
> ```

Thanks @coreydaley for the pointers.
I will try to do that way.

**Comment by apoorvajagtap** [comment]
/close
Closing in favour of #45210

**Comment by divyansh42** [comment]
/close
Another PR https://github.com/openshift/release/pull/45210 created by @apoorvajagtap

**Pull request closed without merging**